src/managers/GoogleCalendarManager.ts

```diff
@@ -0,0 +1,196 @@
+   * messages to the calendar's respective Discord channel for each meeting.
+   * @param client The original client, for access to the configuration.
+   * @param start Date object storing the start of the time window we are querying for.
+   */
+  private async sendMeetingPings(client: BotClient, start: DateTime): Promise<void> {
+    /**
+     * The end of the time window of the query is one minute after the given start time.
+     */
+    const end = start.plus({ minutes: 1 });
+    // Reading in calendar colors from the Google API to color our embeds
+    const colors = await this.calendar.colors.get({});
+    /**
+     * Checking through all calendars in our calendar list...
+     */
+    this.calendarList.map(async calendarID => {
+      const res = await this.calendar.events.list({
+        calendarId: calendarID,
+        timeMin: start.toISO(),
+        timeMax: end.toISO(),
+        singleEvents: true,
+        orderBy: 'startTime',
+      });
+  
+      // Check that there are events to send notifications for
+      if (res && res.data.items) {
+        const events = res.data.items;
+        // Getting the specific color ID of our current calendar to color our embeds later.
+        const calColorResponse = await this.calendar.calendarList.get({ calendarId: calendarID });
+        const calColorID = calColorResponse.data.colorId;
+        events.map(async (event, i) => {
+          if (event && event.start && event.end && event.start.dateTime && event.end.dateTime) {
+            // Send a specific embed for each meeting to the specified channel.
+            const startTime = DateTime.fromISO(event.start.dateTime);
+            const endTime = DateTime.fromISO(event.end.dateTime);
+            const searchInterval = Interval.fromDateTimes(start, end);
+            // We only send embeds for events that are just starting in our time window.
+            if (searchInterval.contains(startTime)) {
+              const mentions = this.calendarMapping[calendarID].getMentions(event);
+              let messageEmbed = new MessageEmbed()
+                .setTitle('🗓️ ' + (event.summary || 'Untitled Event'))
+                .setDescription(event.description || '')
+                .addField('⏰ Time', 
+                  `<t:${Math.trunc(startTime.toSeconds())}:F> to <t:${Math.trunc(endTime.toSeconds())}:F>`)
+                .addField('👥 People', mentions)
+                .setColor('WHITE');
+              // Add the location of the event to the embed if it exists.
+              if (event.location) {
+                messageEmbed = messageEmbed.addField('📍 Location', event.location);
+              }
+              // Update the coloring for our embed.
+              if (event.colorId && colors.data.event) {
+                // Checking if the event has a special color first...
+                messageEmbed = messageEmbed.setColor(colors.data.event[event.colorId].background as ColorResolvable);
```

Comment:
There _could_ be a situation where our ACM calendars have a color that does not match `ColorResolvable`'s enum. Could we consider checking to see if the calendar color it's in that list and then setting another default color if not?

If not, we could also just have the calendar colors match the Discord ones, too. (or this might not even be necessary if Google Calendar doesn't have any extra colors outside of Discord's)

Reply:
From what I've read on ColorResolvable, it accepts any hex number as a color and the Google Calendar colors are in an accepted hex format, so it should work without any issues.